### ISSUE TITLE
Document how to properly create new buttons

### ISSUE THREAD

**Issue by danielo515** [TiddlyWiki/TiddlyWiki5#5379]
**Describe the bug**
Since the beginning of the times, something that ever confused me about tiddlywiki is all the indirect stuff you need to do to properly create a button.

I think it should be clearly documented how a button needs to be created, and what related tiddlers are needed in order for the button to be properly displayed on every possible part of the UI.
Take a look at the example screenshot. What should I do to make the caption and the description appear when the button is under the more toolbar? 

**Screenshots**

![image](https://user-images.githubusercontent.com/2270425/104049301-09e07e00-51e5-11eb-8b2a-9d16f2d7e7b6.png)

My button already has a caption and a description, but I can't guess what else is needed.

**Comment by morosanuae** [comment]
I'm thinking of a customization panel that can help the user to easily modify/create new UI components. But some will say it'll bloat the app. Maybe a plugin?

**Comment by kookma** [comment]
@danielo515 - duplicate one of the core toolbars button! It has special tiddlers for captions, hint, ...

**Comment by danielo515** [comment]
> @danielo515 - duplicate one of the core toolbars button! It has special tiddlers for captions, hint, ...

That is precisely the problem. You can not simply copy a core one because there are other "indirect" tiddlers needed and that are not obvious. I have been developing for TiddlyWiki for 5 years now and this still confuses me, I think it's worth documenting

**Comment by saqimtiaz** [comment]
@danielo515 could you post the code for your button please?

**Comment by gernert** [comment]
@Danielo515

I started with TW5 september 2013 (TW version alpha 10). As you know I am not a programmer.
Since I missed a few things in the layout I got used to in TiddlyWiki Classic, I started with modifying the layout of TW5.
One of the first things I made was a top toolbar with buttons. I documented my findings in my guide TW5 Customization.
At the moment I am updating this guide so only the old one at TiddlySpot (http://tw5custom.tiddlyspot.com/) is available.
All my findings about standard and editor toolbar buttons can be found there.
Note: All new guides and plugins are now hosted at my Raspberry Pi home server at http://www.tongerner.nl/TW5/TW5_index.html

**Comment by Jermolene** [comment]
> I think it should be clearly documented how a button needs to be created, and what related tiddlers are needed in order for the button to be properly displayed on every possible part of the UI.

Yes, it would be great to have some documentation about that. PRs welcome.

**Comment by danielo515** [comment]
> > I think it should be clearly documented how a button needs to be created, and what related tiddlers are needed in order for the button to be properly displayed on every possible part of the UI.
> 
> Yes, it would be great to have some documentation about that. PRs welcome.

I can try to open a PR for that, but first I would need to be sure what is needed.
I look at this information: http://tw5custom.tiddlyspot.com/#Button%20structure%20(standard%20buttons)
And my example button seems to fit all the boxes, but it is not properly displayed when it is on the more toolbar for example. So I need to understand that mechanism before trying to write any docs.

**Comment by saqimtiaz** [comment]
@danielo515 have a look at lines 17-21 here
https://github.com/Jermolene/TiddlyWiki5/blob/master/core/ui/ViewToolbar/new-here.tid#L17

I suspect this is what your toolbar button code might be missing.

It is also difficult to provide feedback when we cannot see your code.

**Comment by danielo515** [comment]
> @danielo515 have a look at lines 17-21 here
> https://github.com/Jermolene/TiddlyWiki5/blob/master/core/ui/ViewToolbar/new-here.tid#L17
> 
> I suspect this is what your toolbar button code might be missing.
> 
> It is also difficult to provide feedback when we cannot see your code.

Thanks for the insight, @saqimtiaz , yeah, my button is not providing that, but I don't see how that can make a difference.
Sorry for not providing the code, here is it: https://github.com/danielo515/TW5-EncryptTiddlerPlugin/blob/master/src/crypt-button.tid

If you want to see it live, here it is: https://danielorodriguez.com/TW5-EncryptTiddlerPlugin/

**Comment by saqimtiaz** [comment]
@danielo515 having looked at your code now I can confirm that is indeed the pattern that you need.  The variable `tv-config-toolbar-text` is set to `yes` inside the more dropdown. 

However given the complexity of your button code and the multiple states, you may need it in more than one place.

For instance try this for $:/plugins/danielo515/encryptTiddler/openPopup:

```
<$reveal state="$:/state/encrypt" type="nomatch" text={{!!title}} animate="no"><$button set="$:/state/encrypt" setTo={{!!title}} class="tc-btn-invisible">{{$:/plugins/danielo515/encryptTiddler/unlocked}}<$list filter="[<tv-config-toolbar-text>match[yes]]">
<span class="tc-btn-text">
<$text text=" Encrypt/Decrypt"/>
</span>
</$list></$button></$reveal><$reveal state="$:/state/encrypt" type="match" text={{!!title}} animate="no"><$button set="$:/state/encrypt" setTo="" class="tc-btn-invisible">{{$:/plugins/danielo515/encryptTiddler/unlocked}}<$list filter="[<tv-config-toolbar-text>match[yes]]">
<span class="tc-btn-text">
<$text text=" Encrypt/Decrypt"/>
</span>
</$list></$button></$reveal>
```

If I were you I would be tempted to create a tiddler called $:/plugins/danielo515/encryptTiddler/unlocked-label with the content:
```
{{$:/plugins/danielo515/encryptTiddler/unlocked}}<$list filter="[<tv-config-toolbar-text>match[yes]]">
<span class="tc-btn-text">
<$text text=" Encrypt/Decrypt"/>
</span>
</$list>
```
and transclude that instead of `$:/plugins/danielo515/encryptTiddler/unlocked` as the button label.

**Comment by saqimtiaz** [comment]
PS: you will need to adapt this to accommodate all the different states of your button but it should point you in the right direction.

**Comment by pmario** [comment]
Question from the OP seems to be answered. So close.
@danielo515 ... If not resolved, then please re-open
